cmd/ebpf/xdp/pdr.h

```diff
@@ -39,10 +39,46 @@ enum outer_header_removal_values {
     OHR_S_TAG_C_TAG = 8,
 };
 
+// Possible optimizations: 
+// 0. Store SDFs in a separate map. PDR will have only id of corresponding SDF.
+// 1. Combine SrcAddress.Type and DstAddress.Type into one __u8 field. Then to retrieve and put data will be used operators & and | .
+// 2. Put all fields into one big structure. Sort in specific order to reduce paddings inside structure.
+
+struct ip_w_mask {
+    __u8 type; // 0: any, 1: ip4, 2: ip6
+    // If type != any, ip field has meaningful value.
+    // If IPv4 -> lower 32 bits. If IPv6 -> all 128 bits.
+    __uint128_t ip;
+    // If type != any, mask field has meaningful value.
+    // If IPv4 mask -> lower 32 bits. If IPv6 mask -> all 128 bits.
+    // Should always be applied to matching ip (except type == any).
+    __uint128_t mask;
+};
+
+struct port_range {
+    __u16 lower_bound; // If not specified in SDF: 0
+    __u16 upper_bound; // If not specified in SDF: 65535
+};
+
+struct sdf_filter {
+    __u8 protocol; // Required by SDF. 0: icmp, 1: ip, 2: tcp, 3: udp
+    struct ip_w_mask src_addr;
+    struct port_range src_port;
+    struct ip_w_mask dst_addr;
+    struct port_range dst_port;
+};
+
+struct sdf_rules {
+    struct sdf_filter sdf_filter;
+    __u32 far_id;
```

Comment:
`__u8 outer_header_removal;` is needed here as well

Reply:
Done?